crypto/s390x_arch.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef S390X_ARCH_H
+# define S390X_ARCH_H
+
+/*
+ * The elements of OPENSSL_s390xcap_P are the doublewords returned by the STFLE
+ * instruction followed by the doubleword pairs returned by instructions' QUERY
+ * functions. If STFLE returns fewer doublewords or an instruction is not
+ * supported, the corresponding element is zero. The order is as follows:
+ *
+ * STFLE:STFLE:STFLE.
+ * KIMD:KIMD:KLMD:KLMD:KM:KM:KMC:KMC:KMAC:KMAC:KMCTR:KMCTR:KMO:KMO:KMF:KMF
+ * :PRNO:PRNO:KMA:KMA
+ */
+# define S390X_STFLE_DWORDS	3
+# define S390X_QUERY_DWORDS	20
+# define S390X_CAP_DWORDS	(S390X_STFLE_DWORDS + S390X_QUERY_DWORDS)
+extern unsigned long long OPENSSL_s390xcap_P[];
+
+/* Offsets */
+# define S390X_STFLE		0
+# define S390X_KIMD		(S390X_STFLE + S390X_STFLE_DWORDS)
+# define S390X_KLMD		(S390X_KIMD + 2)
+# define S390X_KM		(S390X_KLMD + 2)
+# define S390X_KMC		(S390X_KM + 2)
+# define S390X_KMAC		(S390X_KMC + 2)
+# define S390X_KMCTR		(S390X_KMAC + 2)
+# define S390X_KMO		(S390X_KMCTR + 2)
+# define S390X_KMF		(S390X_KMO + 2)
+# define S390X_PRNO		(S390X_KMF + 2)
+# define S390X_KMA		(S390X_PRNO + 2)
+
+/* OPENSSL_s390xcap_P[S390X_STFLE + 2] flags */
+# define S390X_STFLE_VXE	(1ULL << 56)
```

Comment:
I'd like to argue in favour of `(0x8000000000000000 >> 7)` as reflection of the fact that IBM bit numbering starts with most significant bit. Or why not even `(0x8000000000000000 >> (135 % 64))` as explicit reference to documentation? I mean documentation refers to corresponding bit as 135-th. [Same naturally applies to other bits.]

Reply:
And why not
```
#define S390X_CAPBIT(i) 0x8000000000000000 >> (i % 64))
#define S390X_STFLE_VXE S390X_CAPBIT(135)
```

Reply:
I thought about harmonizing this with POP-notations "special" bit numbering. On the other hand, i wanted to document the "normal" bit number so i can directly read the bitmasks to be used with OPENSSL_s390xcap env var.